From 0d9aef522d55ba6681e25f911d42e05bd496053d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:07 -0400
Subject: [usb] cdc-acm: fix I/O after failed open

Message-id: <1410887463-20674-476-git-send-email-dzickus@redhat.com>
Patchwork-id: 95458
O-Subject: [RHEL7 PATCH 475/591] USB: cdc-acm: fix I/O after failed open
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit e4c36076c2a6195ec62c35b03c3fde84d0087dc8
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:44 2014 +0200

    USB: cdc-acm: fix I/O after failed open

    Make sure to kill any already submitted read urbs on read-urb submission
    failures in open in order to prevent doing I/O for a closed port.

    Fixes: 088c64f81284 ("USB: cdc-acm: re-write read processing")
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-acm.c b/drivers/usb/class/cdc-acm.c
index 6c6928a..eddeba6 100644
--- a/drivers/usb/class/cdc-acm.c
+++ b/drivers/usb/class/cdc-acm.c
@@ -506,6 +506,7 @@ static int acm_port_activate(struct tty_port *port, struct tty_struct *tty)
 {
  struct acm *acm = container_of(port, struct acm, port);
  int retval = -ENODEV;
+ int i;
 
  dev_dbg(&acm->control->dev, "%s\n", __func__);
 
@@ -556,6 +557,8 @@ static int acm_port_activate(struct tty_port *port, struct tty_struct *tty)
  return 0;
 
 error_submit_read_urbs:
+ for (i = 0; i < acm->rx_buflimit; i++)
+  usb_kill_urb(acm->read_urbs[i]);
  acm->ctrlout = 0;
  acm_set_control(acm, acm->ctrlout);
 error_set_control:
-- 
1.7.1